sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaMetrics.java

```diff
@@ -39,6 +40,8 @@ public interface KafkaMetrics {
 
   void updateKafkaMetrics();
 
+  void recordBacklogBytes(String topic, int partitionId, long backlog);
```

Comment:
can we just modify already public updateBacklogBytes?

Reply:
We could, where we store all values in a map, and then create gauges afterwards, but we don't need to in this case, we can directly create the gauges. See comments below